python/pyspark/ml/torch/distributor.py

```diff
@@ -548,12 +560,23 @@ def set_torch_config(context: "BarrierTaskContext") -> None:
                 os.environ["NODE_RANK"] = str(context.partitionId())
                 os.environ["RANK"] = str(context.partitionId())
 
-            def set_gpus(context: "BarrierTaskContext") -> None:
-                if CUDA_VISIBLE_DEVICES in os.environ:
-                    return
+            if local_master:
+                # distributed training on a local mode spark cluster
+                def set_gpus(context: "BarrierTaskContext") -> None:
+                    if CUDA_VISIBLE_DEVICES in os.environ:
+                        return
```

Comment:
Shall we need this checking ?

Reply:
I am not sure about this, here follows the distributed mode which respects the `CUDA_VISIBLE_DEVICES` env